### DISCUSSION TITLE
Rspamd/milter integration

### DISCUSSION THREAD

**Discussion by dragoangel** [postalserver/postal#1520] [Feature suggestions]
Hi, it will be cool to switch (or in addition) implement Rspamd.
Rspamd will allow Postal:
1. Total removing of SpamAssassin and scan inbound/outbound emails via same SpamAssassin rules, but in Rspamd, as Rspamd support natively SpamAssassin rules
3. Scan mail over Rspamd rules which have a lot of futures, e.g.: multimaps from redis, http/https lists, combo lists, etc
4. Check SPF,DKIM,DMARC,ARC and check that host have EHLO=A=PTR
5. Handle AV scanning in Rspamd with caching of results, also will extend amount of AV supported
6. Sign DKIM and ARC via key stored at Redis. Postal already use Redis so this cool twice, no additional dependencies. ARC don't supported by Postal, used when one email receive incoming email and send it out somewhere else. E.g. Postal used as MX and use Route to external domain not handled by Postal.

**Comment by dragoangel** [comment]
Oh sorry, yes, no redis is used. And wanted to add that rspamd support dnsbl/rbl which is one of requested futures also at https://github.com/postalserver/postal/discussions/1474

**Comment by adamcooke** [comment]
I haven't looked at rspamd in any detail yet but I'm hoping for it to have a TCP server than I can communicate with from Postal. If it doesn't things are going to be far more complicated. If it does, this is something I can look at this week!

**Reply by dragoangel** [reply]
Yes, there TCP milter server available

**Reply by dragoangel** [reply]
Also quarantine future is possible over rspamd via meta-exporter
